File: app/src/main/java/com/pgssoft/mvvm/services/mock/ObjectMapper.java
package com.pgssoft.mvvm.services.mock;

import com.pgssoft.mvvm.constant.DateFormats;
import com.pgssoft.mvvm.model.api.ApiRate;
import com.pgssoft.mvvm.model.api.ApiTable;
import com.pgssoft.mvvm.model.database.Rate;
import com.pgssoft.mvvm.model.database.Table;
import com.pgssoft.mvvm.models.dto.RateDto;
import com.pgssoft.mvvm.services.interfaces.MapperService;

import org.joda.time.DateTime;
import org.joda.time.format.DateTimeFormat;

/**
 * Created by bstokrocki on 31.01.2017.
 */
public class ObjectMapper implements MapperService {
    @Override
    public Rate map(ApiRate apiApiRate) {
        return new Rate(apiApiRate.getCurrencyCode(), null, apiApiRate.getCurrencyName(), apiApiRate.getRate());
    }

    @Override
    public Table map(ApiTable apiApiTable) {
        return new Table(apiApiTable.getId(), apiApiTable.getNumber(),
                DateTime.parse(apiApiTable.getLastUpdate(),
                DateTimeFormat.forPattern(DateFormats.LAST_UPDATE_DATE_FORMAT)).toDate());
    }

    @Override
    public RateDto map(Rate rate) {
        return new RateDto(rate.getCurrencyCode(), rate.getCurrencyName(), rate.getRate());
    }
}

File: kotlin/app/src/main/java/com/pgssoft/kotlinplayground/service/mock/ObjectMapper.kt
package com.pgssoft.kotlinplayground.service.mock

import com.pgssoft.kotlinplayground.constant.DateFormats
import com.pgssoft.kotlinplayground.model.RateDto
import com.pgssoft.kotlinplayground.model.api.ApiRate
import com.pgssoft.kotlinplayground.model.api.ApiTable
import com.pgssoft.kotlinplayground.model.db.Rate
import com.pgssoft.kotlinplayground.model.db.Table
import com.pgssoft.kotlinplayground.service.iface.MapperService
import org.joda.time.DateTime
import org.joda.time.format.DateTimeFormat

/**
 * Created by bstokrocki on 31.01.2017.
 */
class ObjectMapper : MapperService {
    override fun map(apiRate: ApiRate): Rate {
        return Rate(apiRate.currencyCode, null, apiRate.currencyName, apiRate.rate)
    }

    override fun map(apiTable: ApiTable): Table {
        return Table(apiTable.id, apiTable.number, DateTime.parse(apiTable.lastUpdate,
                        DateTimeFormat.forPattern(DateFormats.LAST_UPDATE_DATE_FORMAT)).toDate())
    }

    override fun map(rate: Rate): RateDto {
        return RateDto(rate.currencyCode, rate.currencyName, rate.rate)
    }
}
